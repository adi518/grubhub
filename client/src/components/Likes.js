import React from 'react';
import styled from 'styled-components';

const StyledLikes = styled.div`
  margin-top: 1rem;
  margin-left: auto;
`;

const Likes = ({ count }) =>
  count ? <StyledLikes> {count} 👍 </StyledLikes> : null;

export default Likes;
