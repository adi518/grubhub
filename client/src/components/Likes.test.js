import React from 'react';
import Likes from './Likes';
import { render } from '@testing-library/react';

describe('Likes component', () => {
  test('without likes', () => {
    const { asFragment } = render(<Likes count={0} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('with likes', () => {
    const { asFragment } = render(<Likes count={2} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
