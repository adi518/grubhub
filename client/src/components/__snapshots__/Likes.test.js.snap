// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Likes component with likes 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa ipDYfv"
  >
     2 👍 
  </div>
</DocumentFragment>
`;

exports[`Likes component without likes 1`] = `<DocumentFragment />`;
